src/otx/core/data/entity/base.py

```diff
@@ -11,29 +11,249 @@
 from typing import Any, Dict, Generic, Iterator, TypeVar
 
 import numpy as np
+import torchvision.transforms.v2.functional as F  # noqa: N812
 from torch import Tensor
+from torch.utils._pytree import tree_flatten
 from torchvision import tv_tensors
-from torchvision.transforms.v2.functional import to_image
 
+from otx.core.data.entity.utils import register_pytree_node
 from otx.core.types.task import OTXTaskType
 
 
-@dataclass
-class ImageInfo:
+class ImageInfo(tv_tensors.TVTensor):
     """Meta info for image.
 
-    :param img_id: Image id
-    :param img_shape: Image shape after preprocessing
-    :param ori_shape: Image shape right after loading it
-    :param pad_shape: Image shape before padding
-    :param scale_factor: Scale factor if the image is rescaled during preprocessing
+    Attributes:
```

Comment:
```suggestion
    Args:
```

Reply:
It was intended. I'd like to notate them as attributes in this class.